### DISCUSSION TITLE
Make the default to bypass

### DISCUSSION THREAD

**Discussion by mankowitz** [cjmellor/approval#9] [Ideas]
For my use case, I would like models to update and create normally and only require approval in certain circumstances. What is the best way to do that?

What I want:
$model->create();  // no approval checking
$model->update(); // no approval checking
$model->turnOnApproval();
$model->update(); // with approval checking
$model->turnOffApproval();
$model->create();  // no approval checking

What I've tried:

I tried making a `MaytBeApproved` trait as follows, without much luck. I also tried to make a `withApproval` method, but that didn't work either. You can see my progress at https://github.com/mankowitz/approval
```
<?php

namespace Cjmellor\Approval\Concerns;

trait MayBeApproved
{

  use MustBeApproved;
  protected bool $requiresApproval = false;

  public function isApprovalBypassed(): bool
  {
    return (!$this->requiresApproval);
  }

  public function requireApproval()
  {
    $this->requiresApproval = true;
  }

  public function withApproval(): static
  {
    $this->bypassApproval = false;
    $this->requiresApproval = true;
    return $this;
  }
}
```

**Comment by cjmellor** [comment]
Hey,

The package comes with a `withoutApproval` method you can chain on before calling an update or create method and it will bypass it. It just returns a Boolean so maybe you could do something with that?

I'm open to any PR's if you have any other ideas.

**Reply by mankowitz** [reply]
My use case is that I've already written tons of CRUD stuff for my various controllers, and now I'd like to add a new module that requires approval for those existing models. I don't want to have to edit all those controllers and add `->withoutApproval` to them. I'd prefer to just add `->withApproval` to the new module. I'll continue to try to debug and submit a PR if I figure it out.

**Reply by mankowitz** [reply]
OK, so I got it to do what I wanted . See #10 . I also added `commit()` and `reject()` methods to update the original tables. I also added a `purge()` method to remove old accepted or rejected approvals.
